### PR TITLE
added clang-tidy rules

### LINKED ISSUE #28: Linting
## Please describe what you would like to see

In order to statically analyze and enforce best practice rules, there should be linter in place - available to run locally and in the CI. 

## Please describe the solution you'd like

[clang-tidy](https://clang.llvm.org/extra/clang-tidy/) seems to be a good option for this. There should be

- a first, initial rule set, 
- a Makefile target , 
- and an integration into the CI
 
implemented.

### PR DESCRIPTION
**Pull request by engelmi**
This PR attemts to fix #28 by introducing 
- clang-tidy as linter
- a set of rules for clang-tidy
- ~integrate it into the CI~ (will be done in the next PR)

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+Checks: >
+    *,
```
Comment (engelmi) [inline .clang-tidy:3]: We are basically using all lint rules available (use `clang-tidy -checks='*' --list-checks` to view all available) and then exclude those we don't want via `-<name>`. 
The following lines contain a list that I excluded so far, e.g. due to `readability-identifier-length` requiring min. 3 chars. If those don't make sense, we can remove them from this blacklist.

**Comment by engelmi** [comment]
~~Waiting for either #29 or #7 to be merged to integrate it into CI and probably think about sth. other than bash scripts to run them.~~

**Comment by mwperina** [comment]
Maybe we can consider GH action to be used, for example: https://github.com/marketplace/actions/c-c-linter

**Review by sdunnagan — APPROVED** [review]
This is awesome!

**Comment by engelmi** [comment]
> Maybe we can consider GH action to be used, for example: https://github.com/marketplace/actions/c-c-linter

I'll open a follow-up PR to integrate it into the workflow. 
Thanks! I'll have a look at it and experiment a bit. Having two actions with clang tools I assume the setup is "duplicated".  Because of this I thought about just calling `make lint` and exchange the format action with a simple `make fmt`.

**Pull request merged**